libraries/lib-stretching-sequence/ClipSegment.h

```diff
@@ -0,0 +1,48 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  ClipSegment.h
+
+  Matthieu Hodgkinson
+
+**********************************************************************/
+#pragma once
+
+#include "AudioSegment.h"
+#include "AudioSegmentSampleView.h"
+#include "PlaybackDirection.h"
+#include "TimeAndPitchInterface.h"
+
+#include <memory>
+
+class ClipInterface;
+using ChannelSampleViews = std::vector<AudioSegmentSampleView>;
+
+class STRETCHING_SEQUENCE_API ClipSegment final :
+    public AudioSegment,
+    public TimeAndPitchSource
```

Comment:
I think the second base could be `private`

Reply:
Could it ? Checking ...
... no, because it passes itself as `TimePitchAndSource` to `mTimeAndPitch`.